discovery/src/main/java/io/airlift/discovery/client/Announcement.java

```diff
@@ -32,7 +33,13 @@
     private final String pool;
     private final Set<ServiceAnnouncement> services;
 
-    public Announcement(String environment, String nodeId, String pool, String location, Set<ServiceAnnouncement> services)
+    @JsonCreator
+    public Announcement(
+            @JsonProperty String environment,
```

Comment:
I don't think we need `@JsonProperty` on the parameter names when using the Jackson parameter names module -- just having `@JsonCreator` should be sufficient. Alternatively, maybe we should add the parameter names to the annotations to be consistent with all the other classes in Airlift?

Reply:
I added the parameter names for consistency.